src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -377,120 +376,157 @@ private String help(MetricRule rule, String unit, String statistic) {
           + " Unit: " + unit;
     }
 
-    private void scrape(List<MetricFamilySamples> mfs) throws CloneNotSupportedException {
-      ActiveConfig config = (ActiveConfig) activeConfig.clone();
-
-      long start = System.currentTimeMillis();
-      for (MetricRule rule: config.rules) {
-        Date startDate = new Date(start - 1000 * rule.delaySeconds);
-        Date endDate = new Date(start - 1000 * (rule.delaySeconds + rule.rangeSeconds));
-        GetMetricStatisticsRequest request = new GetMetricStatisticsRequest();
-        request.setNamespace(rule.awsNamespace);
-        request.setMetricName(rule.awsMetricName);
-        request.setStatistics(rule.awsStatistics);
-        request.setExtendedStatistics(rule.awsExtendedStatistics);
-        request.setEndTime(startDate);
-        request.setStartTime(endDate);
-        request.setPeriod(rule.periodSeconds);
-
-        String baseName = safeName(rule.awsNamespace.toLowerCase() + "_" + toSnakeCase(rule.awsMetricName));
-        String jobName = safeName(rule.awsNamespace.toLowerCase());
-        List<MetricFamilySamples.Sample> sumSamples = new ArrayList<MetricFamilySamples.Sample>();
-        List<MetricFamilySamples.Sample> sampleCountSamples = new ArrayList<MetricFamilySamples.Sample>();
-        List<MetricFamilySamples.Sample> minimumSamples = new ArrayList<MetricFamilySamples.Sample>();
-        List<MetricFamilySamples.Sample> maximumSamples = new ArrayList<MetricFamilySamples.Sample>();
-        List<MetricFamilySamples.Sample> averageSamples = new ArrayList<MetricFamilySamples.Sample>();
-        HashMap<String, ArrayList<MetricFamilySamples.Sample>> extendedSamples = new HashMap<String, ArrayList<MetricFamilySamples.Sample>>();
-
-        String unit = null;
-
-        if (rule.awsNamespace.equals("AWS/DynamoDB")
-                && rule.awsDimensions.contains("GlobalSecondaryIndexName")
-                && brokenDynamoMetrics.contains(rule.awsMetricName)) {
-            baseName += "_index";
+    private static String queryId(MetricRule rule, List<Dimension> dimensions, String stat) {
+      String idComponents = rule.awsNamespace +
+              rule.awsMetricName +
+              dimensions.hashCode() +
+              stat;
+      // We cant't use the concatenated components as the ID directly because AWS will not accept special chars as ID
+      // Also, the ID must start with a letter, not a digit
+      return "q" + Integer.toUnsignedString(idComponents.hashCode());
+    }
+
+    private static MetricDataQuery metricDataQuery(MetricRule rule, List<Dimension> dimensions, String stat) {
```

Comment:
Simple functions like this that are only used once make it hard to piece the code together, it'd be better to inline it. This is part of a chain of at least 5 such functions, requiring me to bounce around the file to try and understand the code.